sql-plugin/src/main/scala/com/nvidia/spark/rapids/RapidsGdsStore.scala

```diff
@@ -64,6 +61,10 @@ class RapidsGdsStore(
     }
   }
 
+  override protected def getMemoryBuffer(buffer: RapidsBufferBase): MemoryBuffer = {
+    buffer.asInstanceOf[RapidsGdsBuffer].materializeMemoryBuffer
```

Comment:
Looking at `RapidsGdsBuffer.materializeMemoryBuffer`, the code uses more state of the parent store than the buffer itself.  That's why I think it makes sense to not have a `materializeBuffer` interface on the buffer but rather handle it in the store directly.  Here's an example that should work after `RapidsGdsBuffer` exposes `fileOffset` as a `val` or otherwise has an accessor to it:
```suggestion
    buffer match {
      case buf: RapidsGdsBuffer =>
        val path = if (buf.id.canShareDiskPaths) {
          sharedBufferFiles.get(buf.id)
        } else {
          buf.id.getDiskPath(diskBlockManager)
        }
        closeOnExcept(DeviceMemoryBuffer.allocate(buf.size)) { devBuffer =>
          CuFile.readFileToDeviceBuffer(devBuffer, path, buf.fileOffset)
          logDebug(s"Created device buffer for $path ${buf.fileOffset}:${buf.size} via GDS")
          devBuffer
        }
      case _ => throw new IllegalArgumentException(s"not a GDS buffer: $buffer")
    }
```

Reply:
Done.